neuro_san/internals/run_context/langchain/llms/default_llm_factory.py

```diff
@@ -62,28 +63,34 @@ class DefaultLlmFactory(ContextTypeLlmFactory, LangChainLlmFactory):
                                     the model description in this class.
     """
 
-    def __init__(self):
+    def __init__(self, config: Optional[Dict[str, Any]] = None):
         """
         Constructor
+
+        :param config: The config dictionary which may or may not contain
+                       keys for the context_type and agent_llm_info_file
         """
         self.llm_infos: Dict[str, Any] = {}
         self.overlayer = DictionaryOverlay()
         self.llm_factories: List[LangChainLlmFactory] = [
             StandardLangChainLlmFactory()
         ]
+        self.config: Dict[str, Any] = config
```

Comment:
Better: Do not store the entire config here, but take just what you need from the constructor.
This way there is guarding from some misguided future temptation to modify the config itself.